File: ios/LocationSharingApp/LocationSharingApp/Models/AuthResult.swift
import Foundation

/**
 * Represents the result of an authentication operation.
 */
struct AuthResult: Codable {
    /// The authenticated user information
    let user: User
    /// The JWT authentication token
    let token: String
    /// The refresh token for token renewal
    let refreshToken: String
    /// The timestamp when the token expires
    let expiresAt: Int64
    
    init(user: User, token: String, refreshToken: String, expiresAt: Int64) {
        self.user = user
        self.token = token
        self.refreshToken = refreshToken
        self.expiresAt = expiresAt
    }
}

File: android/app/src/main/java/com/locationsharing/app/data/models/AuthResult.kt
package com.locationsharing.app.data.models

/**
 * Represents the result of an authentication operation.
 * 
 * @property user The authenticated user information
 * @property token The JWT authentication token
 * @property refreshToken The refresh token for token renewal
 * @property expiresAt The timestamp when the token expires
 */
data class AuthResult(
    val user: User,
    val token: String,
    val refreshToken: String,
    val expiresAt: Long
)